internal/batch/batch_future.go

```diff
@@ -0,0 +1,142 @@
+package batch
+
+import (
+	"fmt"
+	"reflect"
+
+	"go.uber.org/multierr"
+
+	"go.uber.org/cadence/internal"
+)
+
+type BatchFuture interface {
+	internal.Future
+	GetFutures() []internal.Future
+}
```

Comment:
it's probably fine either way, but it's a bit odd to have this both internal and public, and the internal one is essentially unused.

maybe just make `batchFutureImpl` public, and keep (and document) the public one?  it's in `internal/`, it can't be used or referenced externally, should be functionally the same as keeping it private here.

Reply:
Should I use alias then? It still doesn't feel right to expose batchFutureImpl directly.

Reply:
Sure. Sorry I pushed wrong commit ealier that overrode my change